docs(proto): add README (#940)
jstz-dev/jstz: crates/jstz_proto/README.md

## Large Smart Function Deployment

> jstz implements a mechanism to deploy large smart functions that exceed the L1's inbox message size limit (4KB) using a Reveal Data Channel (RDC) approach. The system uses a Merkle tree encoding algorithm to handle data sizes up to approximately 10MB.

Reviewer: ```suggestion
Jstz implements a mechanism to deploy large smart functions that exceed the L1's inbox message size limit (4KB) using a Reveal Data Channel (RDC) approach. The system uses a Merkle tree encoding algorithm to handle data sizes up to approximately 10MB.
```

│  └─────────────────────────┘                                 |           │
│             ↑                      ┌ ─ ─ ─ ─ ─ ┐             |           │
│             └─── 4. preimages ────-            ←──────── preimages       │
│                                    │ local fs  │                         │
│                                    │           │                         │
│                                    └ ─ ─ ─ ─ ─ ┘                         |
└──────────────────────────────────────────────────────────────────────────┘
```

### RevealLargePayload Operation

The core of the large payload handling is the `RevealLargePayload` operation which contains:

- `root_hash`: The root hash of the preimage containing the operation data
- `reveal_type`: The type of operation being revealed (currently supports `DeployFunction`)
- `original_op_hash`: The hash of the original operation being revealed (e.g. hash of `DeployFunction` operation)

While the `RevealLargePayload` operation currently supports only one operation type (`DeployFunction` with large code), its design allows for easy extension to support any type of large payload operation.

### Example Flow

> The protocol assumes the preimages are made available to the kernel before processing the `RevealLargePayload` operation. Only authorized injectors can submit `RevealLargePayload` operations (e.g. jstz node).

Reviewer: ```suggestion
The protocol assumes the preimages are made available to the kernel before processing the `RevealLargePayload` operation. Only authorized injectors such as the Jstz node can submit `RevealLargePayload` operations.
```
